### PR TITLE
add pure-python (async) HTTP request how to

### PR DESCRIPTION
**Pull request by emirkmo**
This PR adds a How-to on making HTTP requests from within pyscript, using pyfetch in Pyodide, which is a wrapper around the `fetch` API. (Read the PR how to if you want to learn more!) The inability to use requests/httpx is one of the first and most common questions by newcomers, hence the need.

This How-to was suggested by https://github.com/pyscript/pyscript/pull/151#issuecomment-1243714175 based on the related community example by me in #151. 

Edit: I verified the output on Sphinx autobuilt docs and on GitHub. It looks good to me!

### CODE REVIEW

File: docs/howtos/http-requests.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+async/await syntax, as concurrent code is preferred for HTTP requests.
+
+The purpose of this guide is not to teach the basics of HTTP requests, but to show how to make them
+from `pyscript` using python, since currently, the common tools such as `requests` and `httpx` are not available.
+
+## Fetch
+
+The `fetch` API is a modern way to make HTTP requests. It is available in all modern browsers, and in Pyodide.
+
+Although there are two ways to use `fetch`, 1) using javacript from pyscript, and 2) using Pyodide's python wrapper,
```
Comment (madhur-tandon) [inline docs/howtos/http-requests.md:17]: `javacript` --> `javascript`

Comment (emirkmo) [reply docs/howtos/http-requests.md:17]: Good call, will do the latter.

Comment (emirkmo) [reply docs/howtos/http-requests.md]: Thanks for the catch and suggestion, I've made the changes. Looks better.

**Comment by emirkmo** [comment]
@madhur-tandon I implemented your review, and made some changes to clarity by expounding upon the use of `pyfetch` and HTTP requests. I also added an API Reference section at the end, for those who don't want to read the whole How-to.

@marimeireles I have finalized the PR with the How-to, please let me know if this is what you had in mind.

File: docs/howtos/http-requests.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+which returns a `FetchResponse` object whenever a request is made. Extra keyword arguments can be passed to `pyfetch`
+which will be passed to the `fetch` API.
+
+The returned object `FetchResponse` has familiar methods and properties
+for dealing with the response, such as `json()` or `status`. See the
+[FetchResponse documentation](https://pyodide.org/en/stable/usage/api/python-api/http.html#pyodide.http.FetchResponse)
+for more information.
+
+# Example
+We will make async HTTP requests to [JSONPlaceholder](https://jsonplaceholder.typicode.com/) fake API using `pyfetch`.
```
Comment (marimeireles) [inline docs/howtos/http-requests.md:35]: We will make async HTTP requests to [JSONPlaceholder](https://jsonplaceholder.typicode.com/)'s fake API using `pyfetch`.

File: docs/howtos/http-requests.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    response = await pyfetch(url, **kwargs)
+    return response
+```
+This function is a wrapper for `pyfetch`, which is a wrapper for the `fetch` API. It is a coroutine function,
+so it must be awaited. It also has type hints, which are not required, but are useful for IDEs and other tools.
+The basic idea is that the `PyScript` will import and call this function, then await the response. Therefore,
+the script containing this function must be importable by `PyScript`.
+
+For this example, we will name the file `request.py` and place it in the same directory as the file
+containing the html code below.
```
Comment (marimeireles) [inline docs/howtos/http-requests.md:75]: code bellow -> code above
(?)

**Review by marimeireles — COMMENTED** [review]
Woh! I love it :D
Thank you so much for the contribution!
Some small nitpicks from my side and another annoying one, we should probably change python -> Python.
Let me know if it's too annoying and I can fix that.
Thank you so much for the great work! Pretty sure it will help lots of folks ❤️

File: docs/howtos/http-requests.md (hunk 3 above)
Comment (emirkmo) [reply docs/howtos/http-requests.md]: Fixed intent

**Comment by emirkmo** [comment]
> Woh! I love it :D Thank you so much for the contribution! Some small nitpicks from my side and another annoying one, we should probably change python -> Python. Let me know if it's too annoying and I can fix that. Thank you so much for the great work! Pretty sure it will help lots of folks heart

Ok, I changed python -> Python or `Python` depending on whether it was in the Tutorial or not. Not sure this is what you asked exactly, but feel free to make changes if you find anything else.

Cheers

File: docs/howtos/http-requests.md (hunk 2 above)
Comment (emirkmo) [reply docs/howtos/http-requests.md]: Clarified what I meant

**Comment by emirkmo** [comment]
@marimeireles anything left blocking approval?  :)

**Comment by marimeireles** [comment]
@emirkmo just me forgetting to click the button.
Thanks for the ping!

**Review by marimeireles — APPROVED** [review]

**Pull request merged**